### PR TITLE
Fix scientificNotation & add reduceRight

### LINKED ISSUE #506: Scientific Notation
There is currently a method called `scientificNotation` that lives in `/lib/scientificNotation.js`.
It is incomplete and needs to be fixed!

This method should take in a float and output a string that can be used in an HTML document of the form `3.25 x 10<sup>2</sup>. It should also take a second argument which specifies the number of significant digits to include.

### Acceptance Criteria:
1. Running `yarn test scientificNotation` results in tests passing.
2. You have written a skeleton method for someone else to work on.
3. You have written tests surrounding your skeleton method.
4. Running `yarn lint` does not print any errors to the console!
5. Optional: write one or two more tests surrounding the method you are solving to account for potential edge cases.

This method can not be worked on until PR #505 is merged.

### Please include the skeleton of a new method + an accompanying test for someone else to work on at the time of creating a pull request! A pull request will most likely be denied if it does not contain a skeleton method for someone else to work on! For more information, please read the [Contributing Guide](https://github.com/BlakeGuilloud/ganon/blob/master/CONTRIBUTING.md).

Thank you so much for your contribution!

### PR DESCRIPTION
**Pull request by showmeyourhits**
Closes #506

- [x] Running `yarn lint` does not trigger any linter errors
- [x] The test of the method you have fixed is passing
- [x] You have written a new skeleton method for someone else to work on!
- [x] You have written tests to accompany your skeleton method

### CODE REVIEW

File: test/scientificNotation.test.js [hunk 1]
```diff
@@ -2,11 +2,11 @@ const { scientificNotation } = require("../lib");
 
 describe("scientificNotation (all valid inputs, positive index)", () => {
   test("3 in scientific notation, no sig figs specified", () => {
-    expect(scientificNotation(3)).toBe("3 x 10<sup>5</sup>");
+    expect(scientificNotation(3)).toBe("3 x 10<sup>0</sup>");
```
Comment (showmeyourhits) [inline test/scientificNotation.test.js]: I know, this can be misleading, but this test are wrong)
Already told about it [here](https://github.com/BlakeGuilloud/ganon/pull/505/files)

File: lib/scientificNotation.js [hunk 2]
```diff
@@ -6,8 +8,17 @@
...
-  throw Error("method is still a skeleton!");
+function scientificNotation(input, sigFigs = 1) {
+  if (!isNumber(input) || typeof input === "string") {
+    throw new Error("input must be a number");
+  }
+  if (!isNumber(sigFigs)) {
+    throw new Error("sigFigs must be a number");
+  }
+
+  var exponential = input.toExponential(sigFigs - 1).split("e");
```
Comment (BlakeGuilloud) [inline lib/scientificNotation.js]: I never knew this was a thing! Nice!
